chia/util/streamable.py

```diff
@@ -69,7 +70,11 @@ def dataclass_from_dict(klass, d):
         return tuple(klass_properties)
     elif dataclasses.is_dataclass(klass):
         # Type is a dataclass, data is a dictionary
-        fieldtypes = {f.name: f.type for f in dataclasses.fields(klass)}
+        metadata = getattr(klass, "_chia_streamable", None)
+        if metadata is not None:
```

Comment:
Why not `isinstance(klass, Streamable)`? Is it so we can use the `_chia_streamable` tag in other non-`Streamable` objects?

Reply:
No, my intent was not to try to work well with not-`Streamable` instances.

I think at one point I just straight up did `klass._chia_streamable` because I thought it was always a `Streamable`...  but then a test failed that used it on a raw `dataclass`.  There are a few places in this change where there is more checking than I feel like I would have designed in but it aligns better with prior behavior.  I could certainly explore cleaning up the conditionals and try to set this all up to be exclusively presuming a proper `Streamable`.

I'll take another look at that unrepresentative test and double check my memory on the uses of `dataclass_from_dict()`.  Unless you want to retain this working for not-`Streamable` instances?  Otherwise the `isinstance()` is reasonable in this case I suppose.  Mostly I try to tend away from it.